### PR TITLE
redirect logs to klog

### PR DESCRIPTION
**Pull request by huww98**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

to be consistent with other k8s projects.

We should migrate to structured and contextual logging gradually, as the k/k repo does.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
log format is switched to klog
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

```docs

```

### CODE REVIEW

**Comment by huww98** [comment]
Added a benchmark. klog is more efficient than logrus. But the redirection causes about 3x overhead. Mostly caused by stack unwind, the call stack between logrus and its hook is too deep.

```
goos: linux
goarch: amd64
pkg: github.com/kubernetes-sigs/alibaba-cloud-csi-driver/pkg/log
cpu: Intel(R) Xeon(R) Platinum
BenchmarkLogrus-4                       	  497246	      2421 ns/op
BenchmarkLogrusStructured-4             	  338302	      3603 ns/op
BenchmarkKlog-4                         	  661293	      1652 ns/op
BenchmarkKlogStructured-4               	  387448	      2936 ns/op
BenchmarkLogrusRedirected-4             	  110848	     10740 ns/op
BenchmarkLogrusStructuredRedirected-4   	  120429	     10234 ns/op
PASS
```
image

We can cut down the overhead by half if we don't want useful backtrace for redirected logs
```
BenchmarkLogrusRedirected-4             	  330480	      3520 ns/op
BenchmarkLogrusStructuredRedirected-4   	  259669	      4505 ns/op
```

File: main.go [hunk 1]
```diff
@@ -95,9 +100,29 @@ type globalMetricConfig struct {
 
 // Nas CSI Plugin
 func main() {
+	csilog.RedirectLogrusToLogr(logrus.StandardLogger(), klog.Background())
```
Comment (mowangdk) [inline main.go:103]: Given that all callers have been switched to klog, what purpose does this function serve?

File: main.go [hunk 2]
```diff
@@ -95,9 +100,29 @@ type globalMetricConfig struct {
...
+
 	flag.Var(features.FunctionalMutableFeatureGate, "feature-gates", "A set of key=value pairs that describe feature gates for alpha/experimental features. "+
 		"Options are:\n"+strings.Join(features.FunctionalMutableFeatureGate.KnownFeatures(), "\n"))
 	flag.Parse()
+
+	// https://github.com/kubernetes/klog/blob/main/examples/coexist_glog/coexist_glog.go
+	// replace this when we remove csi-common
```
[the remark this answers is not in the document]
Comment (huww98) [reply main.go:110]: Yes. I will resolve this one when rebasing.

File: main.go (hunk 1 above)
Comment (mowangdk) [reply main.go:95]: Fine, Is there a possibility we can remove this in the future?

**Pull request merged**